pkg/sharing/files.go

```diff
@@ -257,15 +257,30 @@ func (s *Sharing) ApplyBulkFiles(inst *instance.Instance, docs DocsList) error {
 			}
 			ref = nil
 		}
-		// TODO it's only for directory currently, code needs to be adapted for files
-		doc, err := fs.DirByID(id) // TODO DirOrFileByID
+		dir, file, err := fs.DirOrFileByID(id)
+		// TODO check why the methods (Dir|File)ByID can return a couchdb error or an os error
+		if couchdb.IsNotFoundError(err) {
+			err = os.ErrNotExist
+		}
 		if err != nil && err != os.ErrNotExist {
 			inst.Logger().WithField("nspace", "replicator").
 				Debugf("Error on finding ref of bulk files: %s", err)
 			errm = multierror.Append(errm, err)
 			continue
 		}
-		if ref == nil && doc == nil {
+		if _, ok := target["_deleted"]; ok {
+			if ref == nil || (dir == nil && file == nil) {
+				continue
+			}
+			if dir != nil {
+				err = s.TrashDir(inst, dir)
+			} else {
+				err = s.TrashFile(inst, file)
+			}
```

Comment:
When a directory containing files is deleted, the dir + files are sent in bulk. If a file arrives before the dir here, it's going to be trashed with a ` dir_id: io.cozy.files.trash-dir`, causing unconsistency in the fs.

Reply:
When dir+files are sent in the same bulk, they are sorted so that a file will come after its parent directory. If they are sent in two bulks, the file can trashed before its parent: both will appear in the trash: it looks the other user has first trashed the file and then trashed the directory. It's a tradeoff: we consider this situation as OK (not optimal).